src/main/webapp/app/shared/metis/posting-content/posting-content.components.ts

```diff
@@ -123,6 +126,14 @@ export class PostingContentComponent implements OnInit, OnChanges, OnDestroy {
                         attachmentUnitRefDir +
                         // eslint-disable-next-line @typescript-eslint/no-non-null-asserted-optional-chain
                         this.content!.substring(this.content?.indexOf('(', patternMatch.startIndex)! + 1, this.content?.indexOf(')', patternMatch.startIndex));
+                } else if (ReferenceType.USER === referenceType) {
+                    // referenceStr: string to be displayed for the reference
+                    // eslint-disable-next-line @typescript-eslint/no-non-null-asserted-optional-chain
```

Comment:
How about adding `if (this.content == undefined)` at the start of all this if-else block so we can get rid of these optional chainings?